### PR TITLE
update to nan 2.0

### PR DESCRIPTION
**Pull request by pierreinglebert**
@XhmikosR Just like I feared, nan 2.0 has many breaking changes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "bluebird": "^2.9.25",
     "commander": "^2.8.1",
     "defaults": "^1.0.2",
-    "nan": "^1.8.4",
+    "nan": "2.0.0",
```
Comment (XhmikosR) [inline package.json]: How about using tilde (`~`) here? It's safer compared to `^`.

**Comment by XhmikosR** [comment]
Ouch too many changes :/

**Comment by pierreinglebert** [comment]
I think i covered only 1/3 of the needed changes ^^

File: package.json (hunk 1 above)
Comment (pierreinglebert) [inline package.json]: They are following semver so i prefer to stick with `^`, it's less work to update ^^

**Comment by pierreinglebert** [comment]
it seems to work \o/
@XhmikosR : can you try it ?

**Comment by XhmikosR** [comment]
Seems to work fine from a quick look I had :+1:

**Pull request merged**